> There's also some really useful commands I added in the `new-commands` branch that I'll probably PR into this repo as well, if you wanted to have a look, I'm just not sure exactly what you do and don't want in your main branch, heh.

I took a look a this branch and it looks great! I do prefer /lobotomize over /neuralyze though...
Give me some time to test alllll this cool stuff completely, but for now:

improvements branch:
failure to start
```
Traceback (most recent call last):
  File "/usr/local/bin/oobabot", line 8, in <module>
    sys.exit(main())
  File "/root/staging/src/oobabot/oobabot.py", line 268, in main
    run_cli()
  File "/root/staging/src/oobabot/oobabot.py", line 264, in run_cli
    oobabot.start()
  File "/root/staging/src/oobabot/oobabot.py", line 93, in start
    self.runtime = runtime.Runtime(self.settings)
  File "/root/staging/src/oobabot/runtime.py", line 92, in __init__
    self.prompt_generator = prompt_generator.PromptGenerator(
  File "/root/staging/src/oobabot/prompt_generator.py", line 76, in __init__
    self.bot_prompt_block = self.template_store.format(
  File "/root/staging/src/oobabot/templates.py", line 387, in format
    return self.templates[template_name].format(format_args)
  File "/root/staging/src/oobabot/templates.py", line 415, in format
    return self.template.format(**format_args)
KeyError: 'BOT_SEQUENCE_PREFIX'
```

new-commands branch: (this branch also does not support the prefixes)

failure on catching immersion breaking lines 

![image](https://github.com/jmoney7823956789378/oobabot/assets/133553475/71cfafc4-e795-40f4-9875-cb6663276613)
```
[Reisen]: Ummmm... what were we talking about?
[stoop poops]: hey reisen! tell me what's going on in eintei today[Reisen]:
 Nothing much, it's a pretty quiet day.
[Reisen]: I was just eating some bamboo shoots when you came in and started bothering me :< [Reisen]: Oh! But the flowers2024-04-01 23:56:58,729 ERROR Exception while running <coroutine object DiscordBot._send_text_response_in_channel at 0x730cfcf4c6d0> response: not enough values to unpack (expected 2, got 1)
Stack (most recent call last):
  File "/usr/local/bin/oobabot", line 8, in <module>
    sys.exit(main())
  File "/root/staging/src/oobabot/oobabot.py", line 268, in main
    run_cli()
  File "/root/staging/src/oobabot/oobabot.py", line 264, in run_cli
    oobabot.start()
  File "/root/staging/src/oobabot/oobabot.py", line 100, in start
    asyncio.run(self.runtime.run())
  File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 636, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.10/asyncio/base_events.py", line 603, in run_forever
    self._run_once()
  File "/usr/lib/python3.10/asyncio/base_events.py", line 1909, in _run_once
    handle._run()
  File "/usr/lib/python3.10/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.10/dist-packages/discord/client.py", line 441, in _run_event
    await coro(*args, **kwargs)
  File "/root/staging/src/oobabot/discord_bot.py", line 245, in on_message
    await self._handle_response(
  File "/root/staging/src/oobabot/discord_bot.py", line 311, in _handle_response
    fancy_logger.get().error(
2024-04-01 23:56:58,729 ERROR Ignoring exception in on_message
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/discord/client.py", line 441, in _run_event
    await coro(*args, **kwargs)
  File "/root/staging/src/oobabot/discord_bot.py", line 245, in on_message
    await self._handle_response(
  File "/root/staging/src/oobabot/discord_bot.py", line 316, in _handle_response
    raise task.exception()
  File "/root/staging/src/oobabot/discord_bot.py", line 479, in _send_text_response_in_channel
    last_sent_message = await self._render_streaming_response(
  File "/root/staging/src/oobabot/discord_bot.py", line 616, in _render_streaming_response
    (response, abort_response) = self._filter_immersion_breaking_lines(response)
  File "/root/staging/src/oobabot/discord_bot.py", line 701, in _filter_immersion_breaking_lines
    username, remaining_text = match.groups()
ValueError: not enough values to unpack (expected 2, got 1)
```

Demo'd using this config. removed the suffix stuff when testing on `new-commands` since that feature is not present there.
[aughhh.txt](https://github.com/jmoney7823956789378/oobabot/files/14829669/aughhh.txt)

--- a/src/oobabot/prompt_generator.py
+++ b/src/oobabot/prompt_generator.py
@@ -159,12 +159,10 @@ class PromptGenerator:
         # reverse order
         history_lines = []
 
-        current_datetime = self.get_datetime()
         section_separator = self.template_store.format(
             templates.Templates.SECTION_SEPARATOR,
             {
                 templates.TemplateToken.AI_NAME: self.persona.ai_name,
-                templates.TemplateToken.CURRENTDATETIME: current_datetime,
             },
         )
 

--- a/src/oobabot/templates.py
+++ b/src/oobabot/templates.py
@@ -135,7 +135,6 @@ class TemplateStore:
                 TemplateToken.SYSTEM_SEQUENCE_PREFIX,
                 TemplateToken.SYSTEM_SEQUENCE_SUFFIX,
                 TemplateToken.AI_NAME,
-                TemplateToken.CURRENTDATETIME,
             ],
             "Separator between different sections, if necessary. For example, to "
             "separate example dialogue from the main chat transcript.",
@@ -187,8 +186,6 @@ class TemplateStore:
         ),
         Templates.USER_PROMPT_HISTORY_BLOCK: (
             [
-                TemplateToken.USER_SEQUENCE_PREFIX,
-                TemplateToken.USER_SEQUENCE_SUFFIX,
                 TemplateToken.MESSAGE,
                 TemplateToken.USER_NAME,
             ],
@@ -200,8 +197,6 @@ class TemplateStore:
         ),
         Templates.BOT_PROMPT_HISTORY_BLOCK: (
             [
-                TemplateToken.BOT_SEQUENCE_PREFIX,
-                TemplateToken.BOT_SEQUENCE_SUFFIX,
                 TemplateToken.MESSAGE,
                 TemplateToken.BOT_NAME,
             ],

